homeassistant/components/reolink/__init__.py

```diff
@@ -61,11 +60,15 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     async def async_device_config_update():
         """Update the host state cache and renew the ONVIF-subscription."""
         async with async_timeout.timeout(host.api.timeout):
-            if not await host.update_states():
-                raise UpdateFailed(f"Error updating Reolink {host.api.nvr_name}")
+            try:
+                await host.update_states()
+            except ReolinkError as err
+                raise UpdateFailed(f"Error updating Reolink {host.api.nvr_name}") from err
         async with async_timeout.timeout(host.api.timeout):
-            if not await host.renew():
-                _LOGGER.error("Reolink %s event subscription lost", host.api.nvr_name)
+            try:
+                await host.renew():
+            except ReolinkWebhookException as err:
+                _LOGGER.error("Reolink %s event subscription lost: '%s'", host.api.nvr_name, str(err))
```

Comment:
Shouldn't we raise UpdateFailed here instead of logging an error?

Reply:
Well if the webhook renewal fails we still have the polling every 60 seconds where the motion events are also polled.
So I don't think it is nessesary to mark everything as unavailable just because the pushes will fail.
(Besides motion events more states are polled in a single request to the device)
If the polling (once every 60 seconds) fails, we do raise a UpdateFailed.

Reply:
Won't that mean that we'll spam the log every 60 seconds with this error if the webhook never renews ok?

Reply:
You are right, thanks for pointing this out.
Now, I made a boolean logic to only log this error once.